app.rb

```diff
@@ -1,67 +1,30 @@
 require 'readline'
 require_relative './lib/game.rb'
+require_relative './lib/game_runner.rb'
 require_relative './lib/board/board.rb'
 require_relative './lib/board/board_checker.rb'
 require_relative './lib/players/player.rb'
 require_relative './lib/players/ai.rb'
 require_relative './lib/players/human.rb'
-require_relative './lib/players/player_script.rb'
-require_relative './lib/players/validator.rb'
+require_relative './lib/scripts/player_script.rb'
+require_relative './lib/validators/player_validator.rb'
 require_relative './lib/ui/user_interface.rb'
 require_relative './lib/ui/io.rb'
+require_relative './lib/scripts/game_script.rb'
+require_relative './lib/validators/setup_validator.rb'
+require_relative './lib/game_setup.rb'
 
-def initializer(size, token1, token2)
-  Struct.new('State', :stop_playing, :is_won, :restart)
-  state = Struct::State.new(false, false, false)
+Struct.new('Configuration', :game_progress, :board, :checker, :game_script, :user_interface, :player_validator, :player1, :player2)
+Struct.new('GameProgress', :stop_playing, :is_won, :restart)
```

Comment:
I think it's worth making these structs first class objects, like classes in their own files. The ruby Struct::MyStruct pattern makes them easy to use anywhere, but then you have to be cognizant of run time order of operations. If you create classes in their own files you can just "require" them as needed and not worry about loading the app.rb before any files like game_setup that require these structs.

Reply:
I am with @zacharyjdavy on this.

Plus, `Struct` does not really fit into the language you have created for the domain.

Reply:
I remove these structs and made classes.  Let me know if you are OK with the configuration class.  I created a method to set players later, but it's a little against the spirit of some earlier comments.